### PR TITLE
HAAR-3306: Update approved premises template to account for all domain event types

### PR DESCRIPTION
**Pull request by indyv-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/TemplateHelpers.kt [hunk 1]
```diff
@@ -59,4 +59,6 @@ class TemplateHelpers(
 
   fun buildDateNumber(year: Number?, month: Number?, day: Number?): String =
     buildDate(year?.toInt()?.toString(), month?.toInt()?.toString(), day?.toInt()?.toString())
+
+  fun eq(input: String?, value: String?): Boolean = input == value
```
Comment (mpbostock) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/TemplateHelpers.kt:63]: This only seems to be used in a test, is it still needed? What value does it add over the standard `==`?

Comment (mpbostock) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/TemplateHelpers.kt:63]: Oh, I see it in the mustache template now, sorry

File: src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/TemplateHelpersTest.kt [hunk 2]
```diff
@@ -265,4 +265,19 @@ class TemplateHelpersTest {
...
+  @Nested
+  inner class EqualsTest {
+    @Test
+    fun `eq returns true when args are equal`() {
+      val response = templateHelpers.eq("Value 1", "Value 1")
+      assertThat(response).isTrue()
+    }
+
+    @Test
+    fun `eq returns false when args are equal`() {
```
Comment (mpbostock) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/utils/TemplateHelpersTest.kt:278]: should this title be "eq returns false when args are **not** equal"

do we need any tests for the null cases? are we not able to use the standard `.equals()` in mustache?

**Review by mpbostock — DISMISSED** [review]
LGTM

**Review by mpbostock — APPROVED** [review]
